stdconv/string.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "libmodmqttconv/converter.hpp"
+
+class StringConverter : public DataConverter {
+    public:
+        virtual MqttValue toMqtt(const ModbusRegisters& data) const {
+            std::vector<uint16_t> registers = data.values();
+            ConverterTools::adaptToNetworkByteOrder(registers);
+            size_t size = std::min(mMaxSize, data.getCount() * sizeof(uint16_t));
```

Comment:
I would rather implement the same behaviour as std::string constructed from const char*.
For your real example:

`41 4C 44 31 44 35 46 44 30 30 41 33 41 30 30 00 0B 2A (ASCII: ALD1D5FD00A3A00\u0000)`

Here we have a string of size 16 bytes (including nul) stored in 9 registers (18 bytes), so two last bytes is a garbage. We should stop reading on nul (0x0) and pass it to MqttValue::fromBinary.

MqttValue outputs binary data as std::string, in your case it will be constructed from full 18 bytes including garbage past 0x0 nul character. It may give strange results [as explained here](https://stackoverflow.com/questions/59528273/are-stdstring-with-null-character-possible/59528405#59528405)

Reply:
> Register 14 has only 1 character, so its second byte (0x00) is not part of the content.

It is a mark where string ends.

> MqttValue::fromBinary() gets the max size (here: 15) as argument. If the arg is not set, the content size (here: 16) will be used.

Ok, we can leave maxSize to support some broken devices, but would assume that without it you should read until you see 0x0 and pass all readed bytes to `MqttValue::fromBinary.

Imagine this device holds some user-configurable string in those 14 registers. If user puts 'HELLO' into it, registers should contain:

0x48 0x45 0x4C 0x4C 0x4F 0x0 <garbage or zeroed values in rest>

You should not have to set maxSize to read it - just read until you see 0x0 and return 5 bytes.

Reply:
> you should read until you see 0x0 and pass all readed bytes to `MqttValue::fromBinary.

Your example illustrates a reasonable use case, and users probably won't expect a null byte as part of a string either. I will truncate the string at the first null byte when converting from Modbus to MQTT.

> It is a mark where string ends.

That is true for C-style strings, but the manual of the device has no indication that it also applies to modbus registers. Any unused register of this device returns `0x00`, so zero is apparently the representation of _undefined_ here. Maybe some other device uses `0xFF` for _undefined_, so I chose the maxSize argument to give the user a way to handle this case.

On the other hand, this is a little bit speculative and requires the user to understand and configure everything correctly, so I removed maxSize again. Your assumption (the device uses 0x00 as string terminator) works for my use case and thus I'm fine with that.